### DISCUSSION TITLE
How to serialize nds classes to JSON in java?

### DISCUSSION THREAD

**Discussion by Yesyoor** [ndsev/zserio#539] [Q&A]
Hey,
I am working with zserio and NDS layers. I need to serialize the data from java to JSON. Probably you will wonder, why I try to do what zserio tries to avoid, a JSON text overhead, but I need the data in C# and thus have no other idea about how to pass it to C#.

I want to use the Java application as a standalone application and retrieve the content as JSON.
Unfortunately, the zero overhead of zserio winds up in a huge overhead when trying to serialize the data, due to the many layers and list contained in lists and pointers pointing to other structures, it is a nightmare.

The GSON library fails to serialize the layers due to the contained, self referencing classes inside the layers...
What would be the best option, to get the data out of the generated Java objects?

**Comment by mikir** [comment]
If you need to make JSON export, we recommend to use zserio.runtime.DebugStringUtil.toJsonFile() method implemented directly in [zserio runtime](https://zserio.org/doc/runtime/latest/java/).

**Reply by Yesyoor** [reply]
@mikir 
Thank you, I tried it with:
AdasRoadLocationLayer layerCasted = (AdasRoadLocationLayer) o;
try {
	DebugStringUtil.toJsonFile(layerCasted, "adasRoadLocationLayer.txt");
					
} catch (IOException e) {
	e.printStackTrace();
}
but got this error message:
Walker: Zserio object must have type info enabled (see zserio option -withTypeInfoCode)!
I do not have any influence on how the data is generated, I just receive the binary file and need to deserialize it with zserio and serialize it to json again.

Can I set the TypeInfoCode on the receiver side somehow? I guess it must be set before serializing the data with the zserio SerializeUtil.

**Reply by mikir** [reply]
You should have from somewhere generated Zserio API (Zserio Java objects) compiled from schema using Zserio. These generated API should be compiled with zserio option `-withTypeInfoCode`, please see [Zserio User Guide](https://github.com/ndsev/zserio/blob/master/doc/ZserioUserGuide.md#zserio-command-line-interface).

**Reply by Yesyoor** [reply]
perfect! Thank you already! I will try this and report back.

**Reply by Yesyoor** [reply]
it worked, thanks a lot!

**Comment by Yesyoor** [comment]
The solution recommended by @mikir worked.
